File: backend/calendar/src/main/java/com/example/calendar/domain/calendar/Plan.java
package com.example.calendar.domain.calendar;

import java.time.LocalDateTime;

public class Plan {

    public Long id;

    public Long calendarId;

    public String title;

    public String description;

    public String place;

    public LocalDateTime startTime;

    public LocalDateTime endTime;

    public Plan(Long id, Long calendarId, String title, String description, String place,
                LocalDateTime startTime, LocalDateTime endTime) {
        this.id = id;
        this.calendarId = calendarId;
        this.title = title;
        this.description = description;
        this.place = place;
        this.startTime = startTime;
        this.endTime = endTime;
    }

}

File: frontend/src/domain/Plan.ts
import { format } from '@/utils/date'

export type Plan = {
  id: number
  calendarId: number
  title: string
  description: string
  place: string
  date: Date
  startTime: string
  endTime: string
}

export type PlansByDate = {
  [key: string]: Plan[]
}

export const groupByDate = (plans: Plan[]): PlansByDate => {
  return plans.reduce((acc: { [key: string]: Plan[] }, plan: Plan) => {
    const key = format(plan.date)
    if (key in acc) {
      const returnValue = { ...acc }
      returnValue[key] = [...returnValue[key], plan]
      return returnValue
    }
    return {
      ...acc,
      [key]: [plan],
    }
  }, {})
}
